clang/lib/Sema/SemaTemplate.cpp

```diff
@@ -11698,6 +11698,13 @@ Sema::ActOnTypenameType(Scope *S, SourceLocation TypenameLoc,
     // Construct a dependent template specialization type.
     assert(DTN && "dependent template has non-dependent name?");
     assert(DTN->getQualifier() == SS.getScopeRep());
+
```

Comment:
nit: it would be clearer if we add a comment saying that a typename-specifier can only be followed by an identifier/simple-template-id. (Maybe quote https://eel.is/c++draft/temp.res.general#nt:typename-specifier?)

Reply:
In a function called `ActOnTypenameType`?

Reply:
Oh, I didn't realize it. Never mind.